lib/livebook_web/live/session_live.ex

```diff
@@ -1486,4 +1524,22 @@ defmodule LivebookWeb.SessionLive do
   defp get_page_title(notebook_name) do
     "Livebook - #{notebook_name}"
   end
+
+  defp memory_color(:atom), do: "bg-green-500"
+  defp memory_color(:code), do: "bg-blue-700"
+  defp memory_color(:processes), do: "bg-red-500"
+  defp memory_color(:binary), do: "bg-blue-500"
+  defp memory_color(:ets), do: "bg-yellow-600"
+  defp memory_color(:other), do: "bg-gray-400"
+
+  defp runtime_memory(%{runtime: memory}) do
+    Map.drop(memory, [:total, :system])
+    |> Enum.map(fn {k, v} ->
+      {k, %{unit: format_bytes(v), percentage: Float.round(v / memory.total * 100, 2), value: v}}
+    end)
+  end
+
+  defp uses_memory?(%{runtime: nil}), do: false
```

Comment:
You already have this in other view, maybe move it to session helpers or session :)